lib/plugins/features/movies/controller.js

```diff
@@ -6,5 +6,41 @@ exports.create = (payload) => {
 
   return new Movie().save(payload)
   .then((movie) => new Movie({ id: movie.id }).fetch());
-  
+
 };
+
+exports.display = (params) => {
+
+  if (params === undefined || Object.keys(params).length === 0) {
+    return new Movie().fetchAll();
+  }
+
+  else if (params.release_year !== undefined) {
+    return new Movie().query(function(qb) {
+      qb.where('release_year', '=', params.release_year);
+    })
+    .fetchAll();
+  }
+
+  else if (params.start_year !== undefined && params.end_year !== undefined) {
+    return new Movie().query(function(qb) {
+      qb.where('release_year', '>=', params.start_year).andWhere('release_year', '<=', params.end_year);
```

Comment:
I'm not sure if its possible to filter movies by _just_ start year or _just_ end year. Ideally the final API would have this flexibility. This kind of relates to Stewart's suggestion - you gain a lot more flexibility when you conditionally add queries using Knex.